src/test/java/com/example/bookstore/facades/BookFacadeTest.java

```diff
@@ -0,0 +1,56 @@
+package com.example.bookstore.facades;
+
+import com.example.bookstore.dto.BookDetailVO;
+import com.example.bookstore.dto.BookShortVO;
+import com.example.bookstore.entities.Book;
+import com.example.bookstore.repositories.BookRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.*;
+
+@RunWith(SpringRunner.class)
+public class BookFacadeTest {
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @InjectMocks
+    @Autowired
+    BookFacade bookFacade;
+
+    @Test
+    public void retrieveBookList_shouldReturnAListOfBookShort() {
+        // given
+        List<Book> data = new ArrayList<>();
+        data.add(new Book("A123456789012","title","authors","publisher", "Images", "Book description"));
+        data.add(new Book("B123456789012","title","authors","publisher", "Images", "Book description"));
+        data.add(new Book("C123456789012","title","authors","publisher", "Images", "Book description"));
+        // when
+        Mockito.when(bookRepository.findAll()).thenReturn(data);
+        List<BookShortVO> books = this.bookFacade.retrieveBookList();
+        // then
+        assertEquals(books.size(),data.size());
+        assertEquals(books.get(1).getTitle(),data.get(1).getTitle());
```

Comment:
les get(1) ou get(XX) c'est moche.. Je pensais qu'on ferai ça à la prochaine itération mais on va le faire maintenant ! Ajouter AssertJ regarde comment ça marche tu auras des Matchers un peu plus sympa avec des lambdas et tout et tout pour les collections ;-)